app/src/main/java/org/jtaccuino/app/Studio.java

```diff
@@ -81,15 +81,38 @@ public Parent createParentNode() {
                 ToolBar.createMainToolBar()));
         bp.setBottom(StatusLine.getDefault().getNode());
 
-        sheetPane.getTabs().add(new TabSheet(SheetManager.getDefault().of(notebook)));
+        initTabs();
 
         return bp;
     }
 
-    public void shutdown() {
+    private void initTabs() {
+        var openFiles = FilePersistence.getDefault().openFiles;
+        if (openFiles.isEmpty()) {
+            sheetPane.getTabs().add(new TabSheet(SheetManager.getDefault().of(NotebookPersistence.INSTANCE.of())));
+        } else {
+            openFiles.forEach(file
+                    -> SheetManager.getDefault().open(NotebookPersistence.INSTANCE.of(file.path().toFile()))
+            );
+        }
+    }
+
+    private void shutdown() {
+        var files = FilePersistence.getDefault();
+        files.reset();
         sheetPane.getTabs().stream()
                 .filter(t -> t instanceof TabSheet)
-                .forEach(t -> ((TabSheet) t).close());
+                .map(t -> (TabSheet) t)
+                .filter(t -> null != t.sheet.getNotebook().getFile())
+                .forEach(t -> {
+                    files.add(new FilePersistence.OpenFile(t.sheet.getNotebook().getFile().toPath()));
+                    t.close(false);
+                });
+        SheetManager.getDefault().getRecentFiles().stream().forEach(rf -> {
+            files.add(new FilePersistence.RecentFile(rf.path()));
+        });
+
+        PersistenceManager.writePersistenceFile("files", files);
```

Comment:
strictly speaking only this line is required for storage (and the filled list) on shutdown.
you don't need to use the list from FilePersistence as it will be gone as part of the shutdown anyway.

so you could do:

```suggestion
        PersistenceManager.writePersistenceFile(
            "files",
            SheetManager.getDefault().getRecentFiles().stream()
                .map(rf -> new FilePersistence.RecentFile(rf.path())
                .toList());
```

Reply:
This needs to take into account the open files as well.

Reply:
aah, I see...